packages/vue/src/pages/analytics/analytics-geography/analytics-geography.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <div class="view-wrapper">
+    <analytics-toolbar
+      :show-tabs="true"
+      @tab-change="tabChange($event)"
+    >
+      Geography
+    </analytics-toolbar>
+
+    <sales-map-card :data="salesByStateMarkers" />
+
+    <div class="cards">
+      <revenue-analysis-card :data="salesByStateAndCity" />
+      <revenue-snapshot-card :data="salesByState" />
+    </div>
+  </div>
+  <loading-panel :data="data" />
+</template>
+<script setup lang="ts">
+import { ref } from 'vue';
+// eslint-disable-next-line import/no-unresolved
+import { getSalesByStateAndCity, calcSalesByState } from 'dx-rwa-data';
+
+import { SalesByState, SalesByStateAndCity } from '@/types/analytics';
+import AnalyticsToolbar from '@/pages/analytics/components/analytics-toolbar.vue';
+import LoadingPanel from '../components/loading-panel.vue';
+import SalesMapCard from './components/sales-map-card.vue';
+import RevenueAnalysisCard from './components/revenue-analysis-card.vue';
+import RevenueSnapshotCard from './components/revenue-snapshot-card.vue';
+
+const salesByState = ref<SalesByState | null>(null);
+const salesByStateAndCity = ref<SalesByStateAndCity | null>(null);
+const salesByStateMarkers = ref<Record<string, any> | null>(null);
+
+const data = [salesByStateAndCity, salesByState, salesByStateMarkers];
+
+const createMapCoords = (coords: string) => coords.split(', ').map((coord) => parseFloat(coord));
+
+const loadData = async (startDate: string, endDate: string) => {
+  salesByStateAndCity.value = null;
+  salesByState.value = null;
+  salesByStateMarkers.value = null;
+
+  salesByStateAndCity.value = await getSalesByStateAndCity(startDate, endDate);
+  salesByState.value = calcSalesByState(salesByStateAndCity.value);
+  salesByStateMarkers.value = {
```

Comment:
Move the object to some function. Somethink like getSalesByStateMarkers

Reply:
ok